test/unit/math/prim/prob/discrete_range_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stan/math/prim.hpp>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+TEST(ProbDistributionsDiscreteRange, error_check) {
```

Comment:
The tests need to consider boundary conditions like (lower = 17, upper = 17) and (lower = 17, upper = 18) to make sure we get the right boundary behavior.